contracts/protocol/libraries/logic/MarketplaceLogic.sol

```diff
@@ -516,7 +511,12 @@ library MarketplaceLogic {
 
             // item.token == underlyingAsset but supplied after listing/offering
             // so NToken is transferred instead
-            if (INToken(vars.xTokenAddress).ownerOf(tokenId) == onBehalfOf) {
+            if (INToken(vars.xTokenAddress).ownerOf(tokenId) == address(this)) {
```

Comment:
if from accept bid seems we can still keep this `onBehalfOf` short circuit?
```
if (
                INToken(vars.xTokenAddress).ownerOf(tokenId) == address(this) ||
                INToken(vars.xTokenAddress).ownerOf(tokenId) == onBehalfOf
            )
```

Reply:
no matter if received underlying NFT or NToken it'll fail